assets/js/common/DismissableToast/DismissableToast.test.jsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { toast } from 'react-hot-toast';
+import { screen, render } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+
+import DismissableToast from '.';
+
+describe('DismissableToast component', () => {
+  afterEach(() => {
+    // restore the spy created with spyOn
+    jest.restoreAllMocks();
+  });
+
+  it('renders the text and forward the onclick event with the provided prop id', async () => {
+    const spy = jest.spyOn(toast, 'dismiss');
+    const toastID = 'toast-1';
+    const user = userEvent.setup();
+
+    render(<DismissableToast text="test" id={toastID} />);
+
+    await user.click(screen.getByText('Close'));
+
+    expect(screen.getByText('test')).toBeVisible();
```

Comment:
Isn't this a bit strange?
I would expect that once the Close button is clicked the text not being there

Reply:
I can move the expectation up, but yeah the dismiss is mocked so it's still there xD

Reply:
It is not moving up or down. What is this expectation for?
I mean, i thought that you were testing that when you click `Close` the toast goes away, but you are not.
If this is the case, I would put it above, and maybe even add a comment

Reply:
I'm testing that when we click, the dismiss function of the toast library is called, the dismiss function is spied on to assert that it's called, we don't check if the toast is closed because the library do for us, and because we don't have all the infrastructure in place to dispatch real toasts in the test, so the library is called and this is our assertion and for the component is sufficient enough means that all is wired up

I will add a comment to clarify that, you're right it's misleading

Reply:
I'm alright with the test. It is just that I would move the expect of the text up, so it is easier to understand, that the test doesn't check if the toast goes away when it is clicked (as it is mocked, of course)

Reply:
done